crates/apollo-compiler/src/database/hir_db.rs

```diff
@@ -147,6 +147,9 @@ pub trait HirDatabase: InputDatabase + AstDatabase {
     /// Return all operation fragment spread fields in a corresponding selection set.
     fn operation_fragment_spread_fields(&self, selection_set: SelectionSet) -> Arc<Vec<Field>>;
 
+    /// Return the fields that `selection_set` selects including visiting fragments and inline fragments.
+    fn flattened_operation_fields(&self, selection_set: SelectionSet) -> Vec<Arc<Field>>;
```

Comment:
ooooo yea this is super handy; also handy to have on the hir db as well for a few future purposes